### DISCUSSION TITLE
Functions to ignore uninitialized globals

### DISCUSSION THREAD

**Discussion by Denneisk** [wiremod/wire#2881] [Suggestions]
So
```ts
function func() { print(Var) }

let Var = 123

func()
```

Does not cause a warning.

**Comment by vurvdev** [comment]
I think best thing we could do here is to maybe defer compilation of the body of a function until the first call of it? Would be a bit odd for error feedback, but Zig does this without issue.

I don't think it should be that hard to structure your code to initialize your globals first, though

**Reply by Denneisk** [reply]
Now that I look at it, I think this suggestion is a cope for calling functions before they're defined.
```ts
A = 1
init() # Error: No such function

function init() { print(A) }
```
Testing both cases, TS accepts both ways.
